idaes/core/base/costing_base.py

```diff
@@ -70,6 +70,7 @@ def register_idaes_currency_units():
                 "USD_2006 = 500/499.6 * USD_CE500",
                 "USD_2007 = 500/525.4 * USD_CE500",
                 "USD_2008 = 500/575.4 * USD_CE500",
+                "USD_2008_Nov = 500/566.2 * USD_CE500",
```

Comment:
I understand your point more clearly and appreciate the comment. Let's connect offline to discuss implementation options.

Reply:
declaring global units causes issues. One case: If you import the costing package in two instances and both declare global units.

Reply:
I have talked this over with other members of the core dev team and we agree that these units should not be defined as part of the core unit definition. To prevent issues with re-declaring units, you should write a method that first checks to see if the units are present (and maybe check the conversion factor as well) and only register the units if they do not already exist.

I am looking into ways we can handle this during testing to make sure we do not pollute the global unit registry for the tests with these as well.

Reply:
The following code at the module level of any tests for this code will hopefully work:

```
@pytest.fixture(scope="module", autouse=True)
def _register_units_for_this_module():
    orig = pyo.units.pint_registry()
    tmp = pint_module.UnitRegistry()
    pyo.units.set_pint_registry(tmp)
    yield
    pyo.units.set_pint_registry(orig)
```

This should temporarily switch out the Pyomo unit registry for the duration of tests within the module and then switch it back when it is done (you could also write a context manager to do this). Then, you can register the units you want to use within these tests without having to worry about any conflicting units being defined previously, or having your unit definitions stick around and cause conflicts in later tests.

Reply:
Thank you for providing the code. Would a user be able to similarly define scoped custom currency units within a script, flowsheet or notebook?

Reply:
Yes - IDAES has a function (somewhere) for registering new units which is just a wrapper around `pint`'s own interface (it includes the check for if something is already defined if I recall).

Note that I missed a line above to register your new units, but that should probably occur as part of a test anyway.